### PR TITLE
removed boost

### PR DESCRIPTION
**Pull request by tritonas00**
With this caelum builds without boost, but there is an issue with moving forward the time being very slow.

### CODE REVIEW

**Comment by ohlidalp** [comment]
> but there is an issue with moving forward the time being very slow.

VERY strange, because Caelum actually doesn't use boost at all. It's CMake searches for it, but that's all, no usage in code whatsoever: https://bitbucket.org/ogreaddons/caelum/overview

**Comment by tritonas00** [comment]
Pressing only + works ok. But going fast with shift + lags the game as hell!

shift - is ok though...

**Comment by ohlidalp** [comment]
> Pressing only + works ok. But going fast with shift + lags the game as hell!

@Hiradur @ulteq @mikadou Can any of you guys run this through a profiler, please?

**Comment by Hiradur** [comment]
@tritonas00 I think you mentioned that you also have this problem with caelum from AUR. So is your problem boost-specific or not?

**Comment by tritonas00** [comment]
Aur package without boost (this PR). My PKGBUILD is almost the same with aur.

**Comment by Hiradur** [comment]
So Boost is not the cause here? Can I merge?

**Comment by tritonas00** [comment]
No, the missing boost (this PR) gives this shift + bug

Caelum with boost (without this PR) works fine.

**Comment by Hiradur** [comment]
ok, thanks for the clarification.

**Comment by ulteq** [comment]
I can't reproduce it with Ogre 1.9 (with boost). But I get the same issue with @tritonas00 RoR package.

The time is spent in this method: `Ogre::TerrainQuadTreeNode::notifyDelta`



The issue seems to only occur with both Caelum and Ogre 1.8 compiled without boost.

**Comment by tritonas00** [comment]
> The issue seems to only occur with both Caelum and Ogre **1.8** compiled without boost.

Nope, it occurs with 1.9 boost-less also. I remember tried that.

File: main/src/InternalUtilities.cpp [hunk 1]
```diff
@@ -22,6 +22,7 @@ along with Caelum. If not, see <http://www.gnu.org/licenses/>.
 #include "CaelumExceptions.h"
 #include "InternalUtilities.h"
 #include "PrivatePtr.h"
+#include <cstddef>
```
Comment (mikadou) [inline main/src/InternalUtilities.cpp:25]: What was the reason to add this include?

**Comment by mikadou** [comment]
I'm puzzled I really don't see how the changes in 2171de9b28168dc3b6c10dd320aa53bf18170dc5 can affect the observed behavior in any possible way.
- `find_package` searches for the Boost library in common places and sets some variables if succesful. This shouldn't affect compilation at all.
-  `link_directories` adds directories to be searched when linking against dependencies. Usually this is never needed as it is recommended to link against deps with full paths. Curiosly, in this build system there never happens any linking against boost, so this command basically does nothing.
- `include_directories` adds directories to be searched for when inlcuding headers via include directive. However, there are no boost headers being included anywhere, so again this does basically nothing.

So for me the remaining question is: What is the reason for `#include <cstddef>`?

**Comment by tritonas00** [comment]
without `#include <cstddef>` i get this error

http://pastebin.com/Kfn89KRR

compiling caelum boost free with ogre (also boost free)

**Comment by mikadou** [comment]
Ok I tried the following:
1. Built Ogre 1.9 without boost
2. Built latest ogre-caelum (linked against boostless Ogre 1.9)
3. Applied changes from this PR except for the `#include <cstddef>`
4. Clean build with applied changes (linked against boostless Ogre 1.9)

The include indeed wasn't needed with Ogre 1.9. Furthermore I compared the resulting libCaelum.so files from step 2 and 4. They are identical for every single bit.

I conclude, if there is an issue, it is probably entirely related to Ogre 1.8

**Comment by mikadou** [comment]
@tritonas00  Could you try compiling this branch https://github.com/mikadou/ror-dependencies/tree/boostless_caelum_ogre18 on your system?

This script should build
- Ogre 1.8 without Boost
- Caelum with remove Boost dependency (https://github.com/mikadou/ogre-caelum/tree/boostless)

**Comment by tritonas00** [comment]
Failed building ogre.

`/home/babis/Downloads/ror-dependencies/Source/ogre/RenderSystems/GL/src/atifs/src/ps_1_4.cpp:689:1: error: narrowing conversion of ‘-35051’ from ‘int’ to ‘uint {aka unsigned int}’ inside { } [-Wnarrowing]
 };`

To build ogre 1.8 with gcc 6+ you need `-Wno-narrowing` in CXXFLAGS

I use `CXXFLAGS+=' -Wno-deprecated-declarations -Wno-narrowing'` in my PKGBUILD

Where i can specify that flag in your script?

**Comment by mikadou** [comment]
@tritonas00 you can set the flags like this

```
cmake -DCMAKE_CXX_FLAGS="-Wno-narrowing -Wno-deprecated-declarations" <path-to-sources>
```

The flags should be passed on to all dependencies being built.

**Comment by tritonas00** [comment]
Thanks that worked. Now caelum failed with the same error as above https://github.com/RigsOfRods/ogre-caelum/pull/1#issuecomment-219300307

`/home/babis/Downloads/ror-dependencies/Source/caelum/main/src/InternalUtilities.cpp: In static member function ‘static const String Caelum::InternalUtilities::pointerToString(void*)’:
/home/babis/Downloads/ror-dependencies/Source/caelum/main/src/InternalUtilities.cpp:78:36: error: ‘ptrdiff_t’ does not name a type
         stream << reinterpret_cast<ptrdiff_t>(pointer);`

because `#include <cstddef>` is missing probably

**Comment by mikadou** [comment]
Ok, thanks for verifying. It works for me with gcc5.3. I will try again after updating to gcc6.

**Comment by mikadou** [comment]
@tritonas00 I just updated my system and did a fresh rebuild with gcc 6.1.1. I also had to apply the `-Wno-narrowing -Wno-deprecated-declarations` flags in order to compile Ogre. The strange thing is: I did not get the `ptrdiff_t` error. Caelum was built just fine. Sorry, I'm out of ideas right now ...

**Comment by tritonas00** [comment]
You use caelum source from this repo and both ogre 1.8 and caelum without boost right?

Are you sure ogre compiled without boost?

With boost i don't get that error.

**Comment by mikadou** [comment]
I used exactly this script https://github.com/mikadou/ror-dependencies/tree/boostless_caelum_ogre18

However I'm not 100% sure anymore that Ogre was built without boost. It's not linked dynamically:

```
[fabian@k58 lib (boostless_caelum_ogre18)]$ ldd libOgreMain.so
linux-vdso.so.1 (0x00007ffdf71d8000)
libSM.so.6 => /usr/lib/libSM.so.6 (0x00007f484737a000)
libICE.so.6 => /usr/lib/libICE.so.6 (0x00007f484715d000)
libX11.so.6 => /usr/lib/libX11.so.6 (0x00007f4846e1b000)
libXext.so.6 => /usr/lib/libXext.so.6 (0x00007f4846c08000)
libXt.so.6 => /usr/lib/libXt.so.6 (0x00007f484699f000)
libXaw.so.7 => /usr/lib/libXaw.so.7 (0x00007f484672a000)
libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f484650c000)
libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f4846308000)
libfreeimage.so.3 => /usr/lib/libfreeimage.so.3 (0x00007f4845a67000)
libzzip-0.so.13 => /usr/lib/libzzip-0.so.13 (0x00007f484585e000)
libz.so.1 => /usr/lib/libz.so.1 (0x00007f4845648000)
libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f48452c1000)
libm.so.6 => /usr/lib/libm.so.6 (0x00007f4844fbc000)
libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f4844da6000)
libc.so.6 => /usr/lib/libc.so.6 (0x00007f4844a05000)
libuuid.so.1 => /usr/lib/libuuid.so.1 (0x00007f48447ff000)
libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007f48445dc000)
libXmu.so.6 => /usr/lib/libXmu.so.6 (0x00007f48443c1000)
libXpm.so.4 => /usr/lib/libXpm.so.4 (0x00007f48441ae000)
/usr/lib64/ld-linux-x86-64.so.2 (0x000055792c740000)
libXau.so.6 => /usr/lib/libXau.so.6 (0x00007f4843faa000)
libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007f4843da3000)
```

But checking with `nm libOgreMain.so | c++filt | grep boost` shows there are several boost related symbols in the shared lib. Maybe you could check the output of `nm` on your machine?

**Comment by tritonas00** [comment]
Please remove packages boost and boost-libs (-Rdd if necessary) and try again.

> Maybe you could check the output of nm on your machine?

Recompiling now to check

**Comment by mikadou** [comment]
Yup, after `pacman -Rdd boost intel-tbb` Ogre is really built boost-free. And I get the same missing `ptrdiff_t` error as you @tritonas00.

**Comment by Hiradur** [comment]
If this is an issue with Ogre can we merge it?

**Comment by tritonas00** [comment]
Probably

**Comment by ohlidalp** [comment]
I'm going to merge it and let's research the time issue later. We can always revert the commit if things really don't work out. Furthermore, the issue may be already gone (it appeared to be OGRE-1.8 related).

**Comment by tritonas00** [comment]
>(it appeared to be OGRE-1.8 related)

https://github.com/RigsOfRods/ogre-caelum/pull/1#issuecomment-219282746

**Pull request merged**